crates/cairo-addons/src/vm/hint_definitions/curve.rs

```diff
@@ -0,0 +1,268 @@
+
+            let points_offset = 3 * 2 * N_LIMBS;
+            let q_low_high_high_shifted = calldata[..points_offset].to_vec();
+            let mut calldata_rest = calldata[points_offset..].to_vec();
+
+            // Process 4 arrays of RLC components
+            let mut rlc_components = Vec::<BigUint>::with_capacity((18 + 4 * 2) * N_LIMBS);
+            for _ in 0..4 {
+                let array_len: usize = calldata_rest.remove(0).try_into().map_err(|_| {
+                    HintError::CustomHint("Failed to convert array length to usize".into())
+                })?;
+                let slice_len = min(array_len * N_LIMBS, calldata_rest.len());
+                rlc_components.extend(calldata_rest[..slice_len].iter().cloned());
+                calldata_rest = calldata_rest[slice_len..].to_vec();
+            }
+
+            const EXPECTED_LEN: usize = (18 + 4 * 2) * N_LIMBS;
+            if rlc_components.len() != EXPECTED_LEN {
+                return Err(HintError::CustomHint(
+                    format!(
+                        "Invalid RLC components length: expected {}, got {}",
+                        EXPECTED_LEN,
+                        rlc_components.len()
+                    )
+                    .into(),
+                ));
+            }
+
+            // Fill memory
+            let range_check96_ptr =
+                get_ptr_from_var_name("range_check96_ptr", vm, ids_data, ap_tracking)?;
+            let memory_offset = 4;
+
+            let offset = range_check96_ptr + (4 * N_LIMBS + memory_offset);
+            write_collection_to_addr(offset.unwrap(), &rlc_components, vm)?;
+
+            let offset = range_check96_ptr + (50 * N_LIMBS + memory_offset);
+            write_collection_to_addr(offset.unwrap(), &q_low_high_high_shifted, vm)?;
+            Ok(())
+        },
+    )
+}
+
+pub fn compute_y_from_x_hint() -> Hint {
+    Hint::new(
+        String::from("compute_y_from_x_hint"),
+        |vm: &mut VirtualMachine,
+         _exec_scopes: &mut ExecutionScopes,
+         ids_data: &HashMap<String, HintReference>,
+         ap_tracking: &ApTracking,
+         _constants: &HashMap<String, Felt252>|
+         -> Result<(), HintError> {
+            let a = Uint384::from_var_name("a", vm, ids_data, ap_tracking)?.pack();
+            let b = Uint384::from_var_name("b", vm, ids_data, ap_tracking)?.pack();
+            let p = Uint384::from_var_name("p", vm, ids_data, ap_tracking)?.pack();
+            let g = Uint384::from_var_name("g", vm, ids_data, ap_tracking)?.pack();
+            let x = Uint384::from_var_name("x", vm, ids_data, ap_tracking)?.pack();
+            let v = get_integer_from_var_name("v", vm, ids_data, ap_tracking)?.to_biguint();
+
+            let rhs = (pow(x.clone(), 3) + a * x + b) % p.clone();
+
+            // Currently, only the secp256k1 field is supported
+            let (rhs_felt, g_felt) = if p.to_str_radix(16).to_uppercase() ==
```

Comment:
this check is not present in python

Reply:
We can only support secp256k1 in rust

Reply:
try_get_point_from_x is generalist but can only be used for secp256k1. Maybe a naming problem ?
Why pass the arguments if the curve is already be known. The python version should also add the check maybe?

Note try_get_point_from_x has still an important todo: Add a check for v

Reply:
> Why pass the arguments if the curve is already be known. The python version should also add the check maybe?

The current implementation relying on lambdaworks cryptography requires the `F` Prime order to be known at compile time. The python limitation does not have this limitation. Considering that this is only ever called with the SECP256R1_PRIME_FIELD_ORDER I don't want to make this behavior generic and waste resources on unused code

Reply:
sorry my bad, K1 not R1 - indeed